services/src/main/scala/cromwell/pipeline/service/GitLabProjectVersioning.scala

```diff
@@ -13,38 +13,88 @@ import scala.concurrent.{ ExecutionContext, Future }
 class GitLabProjectVersioning(httpClient: HttpClient, config: GitLabConfig)
     extends ProjectVersioning[VersioningException] {
 
-  override def updateFile(project: Project, projectFile: ProjectFile, version: Option[Version])(
+  private def getLastProjectVersion(project: Project)(implicit ec: ExecutionContext): Future[Option[PipelineVersion]] =
+    getProjectVersions(project).map {
+      case Left(_)                          => None
+      case Right(Seq(v: GitLabVersion, _*)) => Some(v.name)
+      case Right(Seq())                     => Some(PipelineVersion(config.defaultFileVersion))
+    }
+
+  private def compareVersions(
+    optionProjectVersion: Option[PipelineVersion],
+    optionUserVersion: Option[PipelineVersion]
+  )(implicit ec: ExecutionContext): Either[VersioningException, PipelineVersion] =
+    (optionProjectVersion, optionUserVersion) match {
+      case (Some(projectVersion), Some(userVersion)) =>
+        if (projectVersion >= userVersion)
+          Left(
+            VersioningException(
+              s"Your version $userVersion is out of date. Current version of project: $projectVersion"
+            )
+          )
+        else Right(userVersion)
+      case (Some(projectVersion), None) => Right(projectVersion.increaseRevision)
```

Comment:
Why does method `compareVersions` returns increased version? What is the purpose of this method?
Either change method behavior or method name.